### PR TITLE
test: did precompile DOS attack

### PR DESCRIPTION
**Pull request by Moliholy**

### CODE REVIEW

**Review by peterwht — CHANGES_REQUESTED** [review]
Looks great to me! Just a couple reverts to the explicit type declaration and it's good to merge.

File: precompiles/did/src/lib.rs [hunk 1]
```diff
@@ -272,10 +271,8 @@ where
 	fn parse_services(
 		raw_services: Vec<(u8, Bytes)>,
 	) -> Result<BoundedVec<ServiceInfo<R>, R::MaxServices>, PrecompileFailure> {
-		let mut services: BoundedVec<ServiceInfo<R>, R::MaxServices> =
-			BoundedVec::with_bounded_capacity(raw_services.len());
-		let s = raw_services.iter();
-		for service in s {
+		let mut services = BoundedVec::with_bounded_capacity(raw_services.len());
```
[the remark this answers is not in the document]
Comment (Moliholy) [reply precompiles/did/src/lib.rs]: Added clarifying comment regarding the maximum capacity in [49f7f3f](https://github.com/Watr-Protocol/watr/pull/158/commits/49f7f3f6943211523a147538c15c169c92d8fa56).

File: precompiles/did/src/lib.rs [hunk 2]
```diff
@@ -184,15 +184,14 @@ where
 		let origin = Some(R::AddressMapping::into_account_id(handle.context().caller));
 		let did = R::AddressMapping::into_account_id(input.read::<Address>()?.into());
 		let service_details = input.read::<Vec<Bytes>>()?;
-		let mut services: BoundedVec<<R as frame_system::Config>::Hash, R::MaxServices> =
-			BoundedVec::with_bounded_capacity(service_details.len());
+		let mut services = BoundedVec::with_bounded_capacity(service_details.len());
```
[the remark this answers is not in the document]
Comment (Moliholy) [reply precompiles/did/src/lib.rs]: Added clarifying comment regarding the maximum capacity in [49f7f3f](https://github.com/Watr-Protocol/watr/pull/158/commits/49f7f3f6943211523a147538c15c169c92d8fa56).

**Comment by Moliholy** [comment]
I also opened an issue requesting a better interface for `BoundedVec` in https://github.com/paritytech/parity-common/issues/791.

**Review by peterwht — APPROVED** [review]

**Pull request merged**